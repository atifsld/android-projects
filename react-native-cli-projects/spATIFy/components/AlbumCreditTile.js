import React from "react";
import { View, Image, Text } from "react-native";
import { StyleSheet } from "react-native";

function AlbumCreditTile({albumCredit}) {
    return (
        <View style={styles.creditCardView}>
            <View style={styles.creditImageView}>
                <Image style = {styles.creditImage}
                    source={{uri: albumCredit.creditImageUrl}}/>
            </View>
            <View style={styles.creditTextView}>
                <Text numberOfLines={1} style={styles.creditNameText}>{albumCredit.creditName}</Text>
                <Text numberOfLines={1} style={styles.creditContributionText}>{albumCredit.creditContribution}</Text>
            </View>
        </View>
    )
}

const styles = StyleSheet.create({
    creditCardView: {
        flex: 1,
        margin: 8,
        height: 220,
        width: 150,
        backgroundColor: "#64FFDA",
        elevation: 8,
        borderRadius: 8,
        shadowColor: 'black',
        shadowOpacity: 0.25,
        shadowOffset: {width: 0, height: 2},
        shadowRadius: 8
    },
    creditImage: {
        height: 140,
        width: 140,
        borderRadius: 6
    },
    creditImageView: {
        padding: 5,
        elevation: 12,
        alignItems: 'center'
    },
    creditTextView: {
        padding: 8,
        height: 40
    },
    creditNameText: {
        fontSize: 20,
        color: 'black',
        fontWeight: 'bold'
    },
    creditContributionText: {
        fontSize: 16,
        color: 'black',
        fontWeight: 'bold'
    }

})

export default AlbumCreditTile;