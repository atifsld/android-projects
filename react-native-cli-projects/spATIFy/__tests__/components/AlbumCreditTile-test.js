import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import AlbumCreditTile from '../../components/AlbumCreditTile'

const sampleAlbumCredit = {
    "id": "bc288071-4c1a-41ed-9a60-570804da7402",
    "creditAlbumId": "5583e514-cf7b-4214-8397-dfd7e10dfb33",
    "creditName": "Acy R. Lehman",
    "creditContribution": "Design [Cover]",
    "creditImageUrl": "https://i.discogs.com/PkmqjHCGbJZSl6SEgXoi6W1yRGhdEbf9K9Gnugw64hE/rs:fit/g:sm/q:40/h:300/w:300/czM6Ly9kaXNjb2dz/LWRhdGFiYXNlLWlt/YWdlcy9BLTUwNDY5/OC0xMzE4NzYxOTM0/LmpwZWc.jpeg"
}

test('test album credit tile', () => {
    const tree = renderer.create(<AlbumCreditTile albumCredit={sampleAlbumCredit} />).toJSON();
    expect(tree).toMatchSnapshot();
})