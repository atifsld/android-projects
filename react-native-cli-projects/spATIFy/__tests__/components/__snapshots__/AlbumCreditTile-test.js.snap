// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test album credit tile 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#64FFDA",
      "borderRadius": 8,
      "elevation": 8,
      "flex": 1,
      "height": 220,
      "margin": 8,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 8,
      "width": 150,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "elevation": 12,
        "padding": 5,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://i.discogs.com/PkmqjHCGbJZSl6SEgXoi6W1yRGhdEbf9K9Gnugw64hE/rs:fit/g:sm/q:40/h:300/w:300/czM6Ly9kaXNjb2dz/LWRhdGFiYXNlLWlt/YWdlcy9BLTUwNDY5/OC0xMzE4NzYxOTM0/LmpwZWc.jpeg",
        }
      }
      style={
        Object {
          "borderRadius": 6,
          "height": 140,
          "width": 140,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "height": 40,
        "padding": 8,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Object {
          "color": "black",
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Acy R. Lehman
    </Text>
    <Text
      numberOfLines={1}
      style={
        Object {
          "color": "black",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Design [Cover]
    </Text>
  </View>
</View>
`;
